lib/doorkeeper/openid_connect/helpers/controller.rb

```diff
@@ -106,15 +105,17 @@ def reauthenticate_oidc_resource_owner(owner)
         end
 
         def matching_tokens_for_oidc_resource_owner(owner)
-          Doorkeeper::AccessToken.authorized_tokens_for(pre_auth.client.id, owner.id).select do |token|
-            Doorkeeper::AccessToken.scopes_match?(token.scopes, pre_auth.scopes, pre_auth.client.scopes)
-          end
+          AccessToken.matching_token_for(
+            pre_auth.client,
+            owner.id,
+            pre_auth.scopes
+          )
```

Comment:
Hmm this is related to a subtle bug which I'm not sure has been fixed in Doorkeeper: https://github.com/doorkeeper-gem/doorkeeper/issues/1197

The spec mentioned in the issue is still there in `master` / `v5.2.2`: https://github.com/doorkeeper-gem/doorkeeper/blob/v5.2.2/spec/models/doorkeeper/access_token_spec.rb#L544-L548

Do you know anything about this? I'm not sure if this is still relevant, I'll see if we have a spec for this case or add one in `master` if necessary.

Reply:
@linhdangduy ok I had a look, we don't have a spec for this and I decided not to add one, because it wouldn't make much sense to use a different logic here if Doorkeeper would then end up using the other logic later anyway.

So basically, let's ignore this edge-case and revisit if it actually causes problems for someone :wink: 

I also saw `authorizations_controller.rb` defines `matching_token?` with the [same code](https://github.com/doorkeeper-gem/doorkeeper/blob/c075df53caf0c40f5accb276af256858c3aa79eb/app/controllers/doorkeeper/authorizations_controller.rb#L44-L50), so can we replace `oidc_consent_required?` below with simply:

```ruby
!skip_authorization && !matching_token?
```

And it *looks* like `matching_token?` in Doorkeeper should actually do a `.present?` check at the end? Currently it looks like it can return an empty array, which would be truthy even if no matching token is actually found.